compiler-lib/src/dot/implementations.rs

```diff
@@ -94,17 +95,26 @@ impl Dot<Node> for Graph {
                 )
                 .unwrap();
             }
-            for ref_node in node.in_nodes() {
+
+            for (in_idx, pred) in node.in_nodes().enumerate() {
```

Comment:
Gibt bestimmt nen in_nodes_ex dafür

Reply:
this can be found at a lot of places. will fix later.